docs/development.md

```diff
@@ -1,26 +1,254 @@
 # Development
 
-This section outlines how to approach the operations and development of the HTC Grid.
+This section outlines how to develop and deploy a custom application on HTC-Grid.
 
-![Test](./images/development.png)
+At the top level there are 3 main components that need to be developed:
 
-HTC-Grid comes with a mock execution engine mock_compute_engine.cpp. This is the executable that is invoked for every submitted task. In real deployments this executable will need to be replaced with a real implementation. Steps below outline how to modify and apply changes to the custom execution code.
+1. A **client application(s)** that will interact with a deployment of HTC-Grid by submitting tasks and retrieving results.
+2. A **worker lambda function** that will be receiving and executing tasks.
```

Comment:
I'm finding that to some people talking about worker lambda function confuses them... How about "worker function". ??